src/xdsl/dialects/builtin.py

```diff
@@ -104,16 +122,31 @@ class IndexType(ParametrizedAttribute):
 @irdl_attr_definition
 class IntegerAttr(ParametrizedAttribute):
     name = "integer"
-
     value = ParameterDef(IntAttr)
     typ = ParameterDef(AnyOf([IntegerType, IndexType]))
 
     @staticmethod
-    def get(value: int, type: Attribute = IntegerType.get(64)) -> IntegerAttr:
-        return IntegerAttr([IntAttr.get(value), type])
+    @builder
+    def from_int_and_width(value: int, width: int = 64) -> IntegerAttr:
+        return IntegerAttr(
+            [IntAttr.from_int(value),
+             IntegerType.from_width(width)])
 
+    @staticmethod
+    def from_index_int_value(value: int) -> IntegerAttr:
```

Comment:
Shouldn't that be a builder as well? 
How can the framework distinguish this method and the one above? `width` is optional, thus for a call to build with only one parameter of type `int` it should be ambiguous, right?

Reply:
So here, if we would add this, it would never be called anyway because of the previous builder. If there is an ambiguity, the first builder is called.
Also, what I could do is remove the optional parameter, and make it explicit for everyone else. I think this would actually be better.